journals/apps/journals/management/commands/publish_journals.py

```diff
@@ -0,0 +1,162 @@
+        price = options['price']
+        currency = options['currency']
+        access_length = options['access_length']
+        publish = options['publish']
+
+        if not org:
+            raise CommandError('--org <Org Name> must be specified')
+        if not price:
+            raise CommandError('--price <Journal price> must be specified')
+
+        try:
+            journal = Journal.objects.create_journal(name, org, access_length)
+            self._update_wagtail_pages(journal, create=True if name else False, publish=publish)
+
+            journal_meta_data = JournalMetaData(
+                journal=journal,
+                price=price,
+                currency=currency,
+                sku=self._create_sku(journal),
+                title=name,
+                short_desc='{} description'.format(name)
+            )
+        except Exception as err:
+            raise CommandError('Error creating journal: {}'.format(err))
+
+        try:
+            self._update_discovery(
+                journal.organization.site.siteconfiguration.discovery_journal_api_client,
+                journal_meta_data.get_discovery_data()
+            )
+        except HttpClientError as err:
+            # TODO - roll back journal updates if this fails
+            raise CommandError('Error publishing to discovery-service: {}'.format(err.content))
+
+        try:
+            self._update_ecommerce(
+                journal.organization.site.siteconfiguration.ecommerce_journal_api_client,
+                journal_meta_data.get_ecommerce_data()
+            )
+        except HttpClientError as err:
+            # TODO - roll back discovery updates if this fails
+            raise CommandError('Error publishing to ecommerce-service: {}'.format(err.content))
+
+        return journal
+
+    def _update_wagtail_pages(self, journal, create=True, publish=True):
+        '''create/update associated JournalIndexPage and JournalAboutPage'''
+        site = journal.organization.site
+        current_root = Page.objects.get(id=site.root_page_id)
+        index_page = current_root.specific
+
+        # if site root is not a JournalIndexPage, then create one and set it as site root
+        if not isinstance(index_page, JournalIndexPage):
+            # get base root page and add JournalIndexPage as child
+            root_page = Page.get_root_nodes()[0]
+            index_page = JournalIndexPage(
+                title="{} Index Page".format(site.site_name),
+                intro="{} introduction".format(site.site_name)
+            )
+            root_page.add_child(instance=index_page)
+
+            if publish:
+                index_page.save_revision().publish()
+
+            # now set the site root page id to point to new_index_page
+            site.root_page_id = index_page.id
+            site.save()
+
+        # create/update JournalAboutPage
+        journal_about_page = index_page.get_first_child()
+        if not create:
```

Comment:
right, currently it will only get called with create = True. I had this in here in anticipation of calling this function for update as well, where create would be false. But I can remove it until then.

Reply:
It's fine to leave it. I didn't know it was being used for update as well